### PR TITLE
Improve directions when location CRDs not deployed

### PR DESCRIPTION
**Pull request by carlisia**
Wonder if any or all of this would help nudge the user into understanding there's a `step 2` in using our default location/s, which is to deploy the corresponding CRD or any CRD.

### CODE REVIEW

File: pkg/controller/backup_controller.go [hunk 1]
```diff
@@ -273,7 +274,11 @@ func (c *backupController) prepareBackupRequest(backup *api.Backup) *pkgbackup.R
 
 	// validate the storage location, and store the BackupStorageLocation API obj on the request
 	if storageLocation, err := c.backupLocationLister.BackupStorageLocations(request.Namespace).Get(request.Spec.StorageLocation); err != nil {
-		request.Status.ValidationErrors = append(request.Status.ValidationErrors, fmt.Sprintf("Error getting backup storage location: %v", err))
+		if strings.Contains(err.Error(), "not found") {
```
Comment (nrb) [inline pkg/controller/backup_controller.go]: I think I would use `apierrors.IsNotFound(err)` here.

File: pkg/controller/backup_controller.go [hunk 2]
```diff
@@ -273,7 +274,11 @@ func (c *backupController) prepareBackupRequest(backup *api.Backup) *pkgbackup.R
...
+			request.Status.ValidationErrors = append(request.Status.ValidationErrors, fmt.Sprintf("a BackupStorageLocation CRD needs to be deployed before this backup can be created. Error: %v", err))
```
Comment (nrb) [inline pkg/controller/backup_controller.go]: +1 to this message.

**Comment by carlisia** [comment]
I fixed the merge conflict.

File: pkg/controller/backup_controller.go (hunk 2 above)
Comment (skriss) [reply pkg/controller/backup_controller.go]: I'd tweak this to clarify that a BSL *with the name specified in the backup spec* needs to be created/deployed, not just any BSL (similarly to the VSL messages below).  And maybe "before this backup can be executed" rather than "created".

File: pkg/controller/backup_controller.go [hunk 3]
```diff
@@ -309,7 +315,11 @@ func (c *backupController) validateAndGetSnapshotLocations(backup *api.Backup) (
 		// validate each locationName exists as a VolumeSnapshotLocation
 		location, err := c.snapshotLocationLister.VolumeSnapshotLocations(backup.Namespace).Get(locationName)
 		if err != nil {
-			errors = append(errors, fmt.Sprintf("error getting volume snapshot location named %s: %v", locationName, err))
+			if strings.Contains(err.Error(), "not found") {
```
Comment (skriss) [inline pkg/controller/backup_controller.go]: use `apierrors.isNotFound(err)`

File: pkg/controller/backup_controller.go [hunk 4]
```diff
@@ -309,7 +315,11 @@ func (c *backupController) validateAndGetSnapshotLocations(backup *api.Backup) (
...
+				errors = append(errors, fmt.Sprintf("a VolumeSnapshotLocation CRD for the location %s needs to be deployed before this snapshot can be created. Error: %v", locationName, err))
```
Comment (skriss) [inline pkg/controller/backup_controller.go]: Maybe "executed" instead of "created"

File: pkg/controller/backup_sync_controller.go [hunk 5]
```diff
@@ -122,7 +122,7 @@ func shouldSync(location *arkv1api.BackupStorageLocation, now time.Time, backupS
 }
 
 func (c *backupSyncController) run() {
-	c.logger.Info("Checking for backup storage locations to sync into cluster")
+	c.logger.Info("Waiting for existing backup storage locations to sync into cluster")
```
Comment (skriss) [inline pkg/controller/backup_sync_controller.go]: I changed this message to be "Checking for existing backup storage locations...." which I think is more appropriate in all scenarios, including the one where BSLs **do** exist (also changed it to debug, so it's not filling up the log).  So I'd revert this change

**Comment by carlisia** [comment]
I'll make these changes and also rebase.

**Comment by carlisia** [comment]
@skriss updated, ptal when you have a chance.
